library/std/src/io/cursor.rs

```diff
@@ -565,7 +565,7 @@ where
     A: Allocator,
 {
     // For safety reasons, we don't want this sum to overflow ever.
-    // If this saturates, the reserve should panic to avoid any unsound writing.
+    // If this saturates, the reserve should fail to avoid any unsound writing.
```

Comment:
This now reads badly. Perhaps:
```suggestion
    // To avoid unsound writing, if this saturates the reserve should fail.
```

Reply:
I ended up with "If this saturates, the reserve will fail, which will avoid any unsound writing."

Does this suit you?